server/executor/transaction_runner_test.go

```diff
@@ -159,20 +160,26 @@ func runTransactionRunnerTest(t *testing.T, withErrors bool, assert func(t *test
 
 	transactionRun := runner.Run(ctxWithTimeout, transaction, metadata, env)
 
-	done := make(chan bool, 1)
-	subscriptionManager.Subscribe(transactionRun.ResourceID(), subscription.NewSubscriberFunction(
-		func(m subscription.Message) error {
-			tr := m.Content.(model.TransactionRun)
-			if tr.State.IsFinal() {
-				transactionRun = tr
-				done <- true
-			}
-
-			return nil
-		}),
-	)
-	// TODO: this will block indefinitely. we need to set a timeout or something
-	<-done
-
-	assert(t, transactionRun)
+	done := make(chan model.TransactionRun, 1)
```

Comment:
instead of using the channel as a `done` signal, we use it to return the relevant value. by not sharing a variable between 2 goroutines, we eliminate a race condition